cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -2224,6 +2229,10 @@ void RegisterScalarArithmetic(FunctionRegistry* registry) {
           "power_checked", pow_checked_doc);
   DCHECK_OK(registry->AddFunction(std::move(power_checked)));
 
+  // ----------------------------------------------------------------------
+  auto exp = MakeUnaryArithmeticFunctionFloatingPoint<Exp>("exp", exp_doc);
```

Comment:
Perhaps `AddDecimalUnaryKernels<Exp>(exp.get());` could be added here? But since `Decimal128::Exp` doesn't exist we want implicit cast and keep `MakeUnaryArithmeticFunctionFloatingPoint`?